public/app/features/dashboard/components/PanelEditor/state/actions.ts

```diff
@@ -56,9 +57,19 @@ export function exitPanelEditor(): ThunkResult<void> {
 
     const modifiedPanel = getPanel();
     const modifiedSaveModel = modifiedPanel.getSaveModel();
-    const initialSaveModel = getSourcePanel().getSaveModel();
-    const panelChanged = !isEqual(omit(initialSaveModel, 'id'), omit(modifiedSaveModel, 'id'));
-    if (shouldDiscardChanges || !modifiedPanel.libraryPanel || !panelChanged) {
+
+    if (shouldDiscardChanges || !modifiedPanel.libraryPanel) {
+      onConfirm();
+      return;
+    }
+
+    const libraryPanel = await getLibraryPanel(modifiedPanel.libraryPanel!.uid!);
+    const panelChanged = !isEqual(
+      omit(libraryPanel.model, 'id', 'libraryPanel', 'gridPos'),
```

Comment:
I haven't gone through the whole frontend flow (yet), but I think gridpos shouldn't update in the libraryPanel but in the dashboard json:
https://github.com/grafana/grafana/blob/e13333fa86496cede5cf11acce5e5d2f150a89ff/pkg/services/librarypanels/librarypanels.go#L159